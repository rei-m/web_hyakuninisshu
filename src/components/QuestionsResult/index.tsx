import * as React from 'react';
import { Link } from 'react-router-dom';
import styled from 'styled-components';
import { appTheme, withAppTheme } from '../../styles';
import Tatami from '../Tatami';
import QuestionResultsSummary from '../QuestionResultsSummary';

export interface QuestionsResultProps {
  readonly totalCount: number;
  readonly correctCount: number;
  readonly averageAnswerSecond: number;
  readonly onClickRestart: () => void;
}

const RootSection = Tatami.extend`
  width: 100vw;
  min-height: calc(100vh - ${({ theme }) => theme.headerHeight});

  @media screen and (min-width: ${({ theme }) => theme.minWidthWide}) {
    min-height: calc(100vh - ${({ theme }) => theme.headerHeightWide});
  }
`;

const Inner = withAppTheme(styled.div)`
  max-width: 380px;
  padding: ${({ theme }) => theme.spacing2x};
  margin: auto;
`;

const Button = withAppTheme(styled.button)`
  margin: ${({ theme }) => theme.spacing2x};
`;

const QuestionsResult: React.SFC<QuestionsResultProps> = ({
  averageAnswerSecond,
  correctCount,
  totalCount,
  onClickRestart
}) => (
  <RootSection>
    <Inner>
      <QuestionResultsSummary
        title={'正解数'}
        value={`${correctCount} / ${totalCount}`}
        style={{ marginBottom: appTheme.spacing2x }}
      />
      <QuestionResultsSummary
        title={'平均回答時間'}
        value={`${averageAnswerSecond}秒`}
        style={{ marginBottom: appTheme.spacing2x }}
      />
      {correctCount !== totalCount && (
        <Button
          onClick={onClickRestart}
          className="pt-button pt-large pt-icon-repeat"
          data-test="restart"
        >
          間違えた歌の練習をする
        </Button>
      )}
      <Link to="/training" replace={true}>
        <Button
          className="pt-button pt-large pt-icon-undo"
          style={{ marginTop: 0 }}
        >
          メニューに戻る
        </Button>
      </Link>
    </Inner>
  </RootSection>
);

export default QuestionsResult;
