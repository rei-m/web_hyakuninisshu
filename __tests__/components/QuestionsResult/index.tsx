import * as ReactTestRenderer from 'react-test-renderer';
import * as React from 'react';
import { MemoryRouter } from 'react-router-dom';
import { shallow } from 'enzyme';
import QuestionsResult, {
  QuestionsResultProps
} from '../../../src/components/QuestionsResult';
import { sel } from '../../helpers';

describe('<QuestionsResult />', () => {
  let baseProps: QuestionsResultProps;

  beforeEach(() => {
    baseProps = {
      averageAnswerSecond: 2.5,
      correctCount: 8,
      onClickRestart: jest.fn(),
      totalCount: 10
    };
  });

  it('should render component when has wrong', () => {
    const renderer = ReactTestRenderer.create(
      <MemoryRouter>
        <QuestionsResult {...baseProps} />
      </MemoryRouter>
    );
    expect(renderer.toJSON()).toMatchSnapshot();
  });

  it('should render component when has not wrong', () => {
    const props = { ...baseProps, correctCount: 10 };
    const renderer = ReactTestRenderer.create(
      <MemoryRouter>
        <QuestionsResult {...props} />
      </MemoryRouter>
    );
    expect(renderer.toJSON()).toMatchSnapshot();
  });

  it('should fire onClickRestart when restart clicked', () => {
    const wrapper = shallow(<QuestionsResult {...baseProps} />);
    wrapper.find(sel('restart')).simulate('click');
    expect(baseProps.onClickRestart).toHaveBeenCalled();
  });
});
