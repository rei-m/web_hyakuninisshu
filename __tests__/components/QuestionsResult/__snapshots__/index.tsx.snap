// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<QuestionsResult /> should render component when has not wrong 1`] = `
<div
  className="sc-ifAKCX iKgzTM"
>
  <div
    className="sc-EHOje dXuDBg"
  >
    <div
      className="sc-bwzfXH lOjaS"
      style={
        Object {
          "marginBottom": "16px",
        }
      }
    >
      <div
        className="sc-htpNat fCqHKh"
      >
        正解数
      </div>
      <div
        className="sc-bxivhb dTntfs"
      >
        10 / 10
      </div>
    </div>
    <div
      className="sc-bwzfXH lOjaS"
      style={
        Object {
          "marginBottom": "16px",
        }
      }
    >
      <div
        className="sc-htpNat fCqHKh"
      >
        平均回答時間
      </div>
      <div
        className="sc-bxivhb dTntfs"
      >
        2.5秒
      </div>
    </div>
    <a
      href="/training"
      onClick={[Function]}
    >
      <button
        className="bp3-button bp3-large bp3-icon-undo sc-bZQynM ieixNu"
        style={
          Object {
            "marginTop": 0,
          }
        }
      >
        メニューに戻る
      </button>
    </a>
  </div>
</div>
`;

exports[`<QuestionsResult /> should render component when has wrong 1`] = `
<div
  className="sc-ifAKCX iKgzTM"
>
  <div
    className="sc-EHOje dXuDBg"
  >
    <div
      className="sc-bwzfXH lOjaS"
      style={
        Object {
          "marginBottom": "16px",
        }
      }
    >
      <div
        className="sc-htpNat fCqHKh"
      >
        正解数
      </div>
      <div
        className="sc-bxivhb dTntfs"
      >
        8 / 10
      </div>
    </div>
    <div
      className="sc-bwzfXH lOjaS"
      style={
        Object {
          "marginBottom": "16px",
        }
      }
    >
      <div
        className="sc-htpNat fCqHKh"
      >
        平均回答時間
      </div>
      <div
        className="sc-bxivhb dTntfs"
      >
        2.5秒
      </div>
    </div>
    <button
      className="bp3-button bp3-large bp3-icon-repeat sc-bZQynM ieixNu"
      data-test="restart"
      onClick={[MockFunction]}
    >
      間違えた歌の練習をする
    </button>
    <a
      href="/training"
      onClick={[Function]}
    >
      <button
        className="bp3-button bp3-large bp3-icon-undo sc-bZQynM ieixNu"
        style={
          Object {
            "marginTop": 0,
          }
        }
      >
        メニューに戻る
      </button>
    </a>
  </div>
</div>
`;
